generator-annotations/src/main/java/io/fabric8/crd/generator/annotation/AdditionalPrinterColumn.java

```diff
@@ -22,27 +22,74 @@
 import java.lang.annotation.Target;
 
 /**
- * Defines an additional printer column. Must be placed at the root of the custom resource.
+ * Defines an additional printer column. Must be placed at the root of the
+ * custom resource.
  */
 @Repeatable(AdditionalPrinterColumns.class)
 @Target({ ElementType.TYPE })
 @Retention(RetentionPolicy.RUNTIME)
 public @interface AdditionalPrinterColumn {
 
+  //https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#type
+  enum Type {
+
+    STRING("string"),
+    INTEGER("integer"),
+    NUMBER("number"),
+    BOOLEAN("boolean"),
+    DATE("date");
+
+    public final String value;
+
+    Type(String value) {
+      this.value = value;
+    }
+
+    public String getValue() {
+      return value;
+    }
+  }
+
+  // https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#format
+  enum Format {
+
+    NONE(""),
+    INT32("int32"),
+    INT64("int64"),
+    FLOAT("float"),
+    DOUBLE("double"),
+    BYTE("byte"),
+    BINARY("binary"),
```

Comment:
Sorry, havn't checked it twice: `binary` is not allowed. Only `byte`.

https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#format

Reply:
Updated. Altenatively, I'm fine if you want to take over ownership of this feature from your branch.

Reply:
If you want, you can assign it to me but it doesn't make a lot sense too me as yours is almost ready.
Your approach is able collect the path, my approach tries to distinguesh between top level annotations and annotations in the hierarchy a little bit more but without the ability to collect the path.
So it depends on which direction we want to go.